src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -3667,13 +3667,7 @@ trait Types
 
       if (suspended) tp =:= origin
       else if (instValid) checkIsSameType(tp)
-      else isRelatable(tp) && {
-        val newInst = wildcardToTypeVarMap(tp)
```

Comment:
The only rationale for the type vars that I can come up with is that they, unlike wildcards, can/will make sure that they can be instantiated to a valid type. A wildcard will just accept any comparison to another type, while a type var tracks the comparisons and in the end makes sure the accumulated lower bounds are compatible with the accumulated upper bounds. Maybe that check isn't happening properly, and thus the typevars are indistinguishable from wildcards, but I'd like to explore that hypothesis more (or if you did, record the exploration) before dropping them.

Reply:
I'm having trouble convincing myself this is actually important, since each wildcard is instantiated to a fresh type var -- so, how can they ever accumulate incompatible bounds?

Reply:
Hmm, good point. I guess it depends on the call to `constr isWithinBounds newInst`, but when are those new type variables ever instantiated?

Reply:
@adriaanm WDYT about this: I make a PR with an assertion that we never instantiate type variables with wildcard origin and we let CI check that. Then we can run a community build, right? I don't know another way to check this. Here it is: #8283

Reply:
```
    def isWithinBounds(tp: Type): Boolean = (
         lobounds.forall(_ <:< tp)
      && hibounds.forall(tp <:< _)
      && (numlo == NoType || (numlo weak_<:< tp))
      && (numhi == NoType || (tp weak_<:< numhi))
    )
```

is kind of a lightweight instantiate? Trying to reverse the intent here, not sure.

Reply:
You mean on the type which was the result of `wildcardToTypeVarMap`. Sure but this would only accumulate bounds in the freshly baked type variables. My hypothesis is that we never instantiate them and the accumulated bounds don't matter.

Reply:
yeah, so maybe we should run a similar check on all freshly created type vars?

Reply:
Sorry, I'm not being very clear. After we check `isWithinBounds`, we should probably check that there's actually a solution for the constraints implied by freshly created type vars, now that they've been constrained. If we don't do that, I agree it's equivalent to the cleanup in your PR.

Reply:
So the question becomes: *should* we indeed check that they can be instantiated? I don't know.

Reply:
Ah I see, I did not think of that. Makes sense, we can try that too +1:

Reply:
Thanks for investigating! I know it can get a bit annoying to figure out the original intent in this part of the code base :-)

Reply:
Actually I couldn't find an easy way to instantiate the type variables at this point. `solve` expects also a list of type parameters and other methods which I could use are in `Infer` and not accessible here.